File: java/07 - Demo 3 - Refactoring/src/main/java/movemethod/Account.java
package movemethod;

public class Account {
	private AccountType type;

	private int daysOverdrawn;

	public Account(AccountType type, int daysOverdrawn) {
		super();
		this.type = type;
		this.daysOverdrawn = daysOverdrawn;
	}

	public double bankcharge() {
		double result = 4.5;

		result += overdraftCharge();

		return result;
	}

	private double overdraftCharge() {
		if (type.isPremium()) {
			double result = 10;
			if (getDaysOverdrawn() > 7)
				result += (getDaysOverdrawn() - 7) * 1.0;
			return result;
		} else
			return getDaysOverdrawn() * 1.75;
	}

	public int getDaysOverdrawn() {
		return daysOverdrawn;
	}

}

File: csharp/07 - Demo 3 - Refactoring/Refactoring/MoveMethod/Account.cs
namespace Refactoring.MoveMethod
{
    public class Account
    {
        private readonly AccountType _type;

        public Account(AccountType type, int daysOverdrawn)
        {
            _type = type;
            DaysOverdrawn = daysOverdrawn;
        }

        public int DaysOverdrawn { get; }

        public double Bankcharge()
        {
            var result = 4.5;

            result += OverdraftCharge();

            return result;
        }

        private double OverdraftCharge()
        {
            if (_type.IsPremium)
            {
                double result = 10;
                if (DaysOverdrawn > 7)
                    result += (DaysOverdrawn - 7) * 1.0;
                return result;
            }
            return DaysOverdrawn * 1.75;
        }
        
    }
}